pyscf/scf/hf.py

```diff
@@ -719,13 +721,38 @@ def fproj(mo):
         dm = lib.tag_array(dm, mo_coeff=mo_coeff[:,::-1], mo_occ=mo_occ)
     return dm
 
+def init_guess_by_vsap(mol, sapgrasp='small'):
+    '''Generate initial guess density matrix using the superposition of
+    atomic potentials (SAP) guess, doi:10.1021/acs.jctc.8b01089
+
+    Args:
+        mol : MoleBase object
+            the molecule object for which the initial guess is evaluated
+        sapgrasp : str (small | large)
+            whether to use small or large coefficient set
+
+    Returns:
+        dm0 : ndarray
+            SAP initial guess density matrix
+    '''
+    Vsap = make_vsap(mol, sapgrasp=sapgrasp)
+
+    hcore = get_hcore(mol)
+    s = get_ovlp(mol)
+    e, coeff = eig(hcore + Vsap, s)
+
+    mf = RHF(mol)
+    occ = get_occ(mf, e, coeff)
+
+    dm = make_rdm1(coeff, occ)
+    return dm
 
 def get_init_guess(mol, key='minao', **kwargs):
     '''Generate density matrix for initial guess
 
     Kwargs:
         key : str
-            One of 'minao', 'atom', 'huckel', 'hcore', '1e', 'chkfile'.
+            One of 'minao', 'atom', 'huckel', 'hcore', '1e', 'vsap', 'chkfile'.
```

Comment:
Isn't `vsap` already used in the DFT code for the original variant?

Reply:
Well, yes, but the PySCF logic seems to call the corresponding vsap function depending on whether the SCF object is of type DFT or HF. (well, it used to only call the DFT one, but I have now added a check which should take care of calling the corresponding vsap function depending on whether the object is DFT or HF).